Add processor docs to Elastic Agent settings reference (#500)
elastic/observability-docs: docs/en/ingest-management/elastic-agent/configuration/processors/processor-add_observer_metadata.asciidoc

[[add_observer_metadata-processor]]
== Add Observer metadata

++++
<titleabbrev>add_observer_metadata</titleabbrev>
++++

beta[]

[source,yaml]
-------------------------------------------------------------------------------
processors:
  - add_observer_metadata:
      cache.ttl: 5m
      geo:
        name: nyc-dc1-rack1
        location: 40.7128, -74.0060
        continent_name: North America
        country_iso_code: US
        region_name: New York
        region_iso_code: NY
        city_name: New York
-------------------------------------------------------------------------------

It has the following settings:

`netinfo.enabled`:: (Optional) Default true. Include IP addresses and MAC addresses as fields observer.ip and observer.mac

> `cache.ttl`:: (Optional) The processor uses an internal cache for the observer metadata. This sets the cache expiration time. The default is 5m, negative values disable caching altogether.

Reviewer: ```suggestion
`cache.ttl`:: (Optional) The processor uses an internal cache for the observer metadata. This sets the cache expiration time. The default is `5m`. Negative values disable caching altogether.
```

> `geo.name`:: (Optional) User definable token to be used for identifying a discrete location. Frequently a datacenter, rack, or similar.

Reviewer: ```suggestion
`geo.name`:: (Optional) User definable token to be used for identifying a discrete location. Frequently a data center, rack, or similar.
```

> `geo.location`:: (Optional) Longitude and latitude in comma separated format.

Reviewer: ```suggestion
`geo.location`:: (Optional) Longitude and latitude in comma-separated format.
```
